.github/workflows/publish.yml

```diff
@@ -0,0 +1,28 @@
+name: Build and publish
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build with bikeshed
+        run: |
+          git clone --depth 1 https://github.com/tabatkins/bikeshed.git
+          pip3 install --user --editable ./bikeshed
+          export PATH="$(python3 -m site --user-base)/bin:${PATH}"
+          bikeshed update
+          mkdir out
+          ./compile.sh
+
+      - name: Deploy to GitHub Pages (only on main branch)
+        if: ${{ success() && github.ref == 'refs/heads/main' }}
```

Comment:
what does `github.ref` do in case of PRs? we don't want stuff to be published until it lands

Reply:
I't will be `refs/pull/1234/merge`, see https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request